cli/cmd/main.go

```diff
@@ -895,27 +889,26 @@ func createDataFilter(c *cli.Context) (*datapb.Filter, error) {
 	if len(c.StringSlice(dataFlagMimeTypes)) != 0 {
 		filter.MimeType = c.StringSlice(dataFlagMimeTypes)
 	}
-	if c.StringSlice(dataFlagTags) != nil && c.String(dataFlagTaggedUntagged) != "" {
-		return nil, errors.New("cannot filter based on matching and tagged or untagged")
-	}
 	if c.StringSlice(dataFlagTags) != nil {
-		filter.TagsFilter = &datapb.TagsFilter{
-			Type: datapb.TagsFilterType_TAGS_FILTER_TYPE_MATCH_BY_OR,
-			Tags: c.StringSlice(dataFlagTags),
-		}
-	}
-	if c.String(dataFlagTaggedUntagged) != "" {
-		switch c.String(dataFlagTaggedUntagged) {
+		tagsLen := len(c.StringSlice(dataFlagTags))
+		switch c.StringSlice(dataFlagTags)[0] {
 		case "tagged":
-			filter.TagsFilter = &datapb.TagsFilter{
-				Type: datapb.TagsFilterType_TAGS_FILTER_TYPE_TAGGED,
+			if tagsLen == 1 {
```

Comment:
What happens if tagsLen != 1? It looks like it might silently fail or behave in unexpected ways (however our backend responds if no filter is passed)

Reply:
Rather than using a switch, can check the length + tagged/untagged  and return early, then default to the below filter.